azurerm/internal/services/eventgrid/tests/eventgrid_domain_resource_test.go

```diff
@@ -250,3 +270,26 @@ resource "azurerm_eventgrid_domain" "test" {
 }
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger)
 }
+
+func testAccAzureRMEventGridDomain_premium(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_eventgrid_domain" "test" {
+  name                = "acctesteg-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku {
```

Comment:
We'll want to change this around a bit.
```suggestion
  sku = "Premium"
```